### PR TITLE
test/e2e: add test for rejecting symlinked cache.

### PR DESCRIPTION
**Pull request by klihub**

### CODE REVIEW

File: test/e2e/policies.test-suite/topology-aware/n4c16/test00-reject-symlink/code.var.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# This test verifies that nri-resmgr properly refuses to start if a cache
```
Comment (jukkar) [inline test/e2e/policies.test-suite/topology-aware/n4c16/test00-reject-symlink/code.var.sh]: better to name the test to something else than "test00" as there is already a test00, better to use test13 (next available free number)

**Review by jukkar — CHANGES_REQUESTED** [review]

File: test/e2e/policies.test-suite/topology-aware/n4c16/test00-reject-symlink/code.var.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+)
+status=$?
+
+# Check and report test status.
+case "$status" in
+    1) error "ERROR: nri-resmgr expected to reject symlinked cache, but it did not.";;
+    2) error "ERROR: nri-resmgr failed to start, but looks like for the wrong reason...";;
+    0) echo "OK: nri-resmgr properly rejected symlinked cache"; return 0;;
+    *) error "ERROR: test failed with unexpected status.";;
+esac
```
Comment (fmuyassarov) [inline test/e2e/policies.test-suite/topology-aware/n4c16/test00-reject-symlink/code.var.sh]: If the status is return code, how do we search here for a specific string? I mean, $status is either 0 or 2 and not a part of a log string.

**Review by fmuyassarov — COMMENTED** [review]
I left one question and apart from Jukka's comment, LGTM

File: test/e2e/policies.test-suite/topology-aware/n4c16/test00-reject-symlink/code.var.sh (hunk 2 above)
Comment (klihub) [reply test/e2e/policies.test-suite/topology-aware/n4c16/test00-reject-symlink/code.var.sh]: > If the status is return code, how do we search here for a specific string? I mean, $status is either 0 or 2 and not a part of a log string.

```
# Try to re-launch nri-resmgr, check whether and how it fails.
(
  trap 'restore_cache' 0
  if (wait_t=5 launch nri-resmgr); then
      exit 1
  else
      if ! vm-command "kubectl -n kube-system logs daemonset.apps/nri-resmgr | \
          grep -q 'exists, but is a symbolic link'"; then
          exit 2
      else
          exit 0
      fi
  fi
)
status=$?
```
So in a subshell (`(trap ...)`) we try to start up the `nri-resmgr` daemonset and exit with 1, 2, or 0 depending whether `nri-resmgr` succeeds to start, fails to start but for the wrong reason (`"exists, but is a symbolic link"` not found in logs), or failes/refuses to start up for the right reason (that message of interest found in logs). `$status` is set to this same exit code. In the end we check that code and terminate the test case accordingly.

Comment (fmuyassarov) [reply test/e2e/policies.test-suite/topology-aware/n4c16/test00-reject-symlink/code.var.sh]: thanks understood now, I overlooked at 1), 2), 0), *. LGTM

**Review by fmuyassarov — APPROVED** [review]
LGTM

**Review by jukkar — APPROVED** [review]

**Pull request merged**